src/types.ts

```diff
@@ -31,15 +31,15 @@ export type HttpOperationTransformer<T> = (opts: T) => IHttpOperation;
 
 export type ArrayCallbackParameters<T> = [T, number, T[]];
 
-export type AvailableContext = 'service' | 'path' | 'operation';
+export type AvailableContext = 'service' | 'path' | 'operation' | 'callback';
 
 export type References = Record<string, { resolved: boolean; value: string }>;
 
 export type TransformerContext<T extends Fragment = Fragment> = {
   document: T;
   context: AvailableContext;
-  parentId: string;
-  readonly ids: Record<AvailableContext, string>;
+  parentId: string | undefined;
+  readonly ids: Partial<Record<AvailableContext, string>>;
```

Comment:
```suggestion
  readonly ids: Omit<Record<AvailableContext, string>, 'callback'>;
```
perhaps we do it this way to better reflect the actual value ids may hold?